### ISSUE TITLE
Bug: the  v1.5.0 iframe not support local source ? 

### ISSUE THREAD

**Issue by MarkMindCkm** [obsidianmd/obsidian-api#142]
**Steps to reproduce:**
- when plugin use iframe , but iframe not support load local source ,for example css , js and images 
- ![11111](https://github.com/MarkMindCkm/obsidian-markmind-private/assets/18719494/b3b980ea-beef-4915-a04d-8b9a236c6788)

**Comment by lishid** [comment]
Yeah, we had to block these because they were a security vulnerability. What URL/origin is your iframe using?

**Comment by MarkMindCkm** [comment]
> Yeah, we had to block these because they were a security vulnerability. What URL/origin is your iframe using?

I put a local source   to `.obsidian`  of vault , 
then use  `this.app.vault.adapter.getResourcePath('.obsidian/web/viewer.html');` to create a iframe url ,
 Css and js are referenced in the iframe

**Comment by lishid** [comment]
Let me think about it - unfortunately the vulnerability involves loading a local html file into an iframe using which it would be able to read arbitrary local files through the app:// URIs.

**Comment by lishid** [comment]
Hmm one workaround for now I can think of is to read the files and send them directly to the iframe. Alternatively you could embed them into the HTML. Both approaches are kinda tedious though.

**Comment by MarkMindCkm** [comment]
> Hmm one workaround for now I can think of is to read the files and send them directly to the iframe. Alternatively you could embed them into the HTML. Both approaches are kinda tedious though.

Indeed, but resources such as fonts and images still cannot be processed .

Additionally, I think that there are potential issues with using plugins . The plugin can access system resources .

**Comment by MarkMindCkm** [comment]
Can it be blocked in safe mode and allowed to load resources when the plugin is enabled ?

**Comment by jiangnanqw12** [comment]
> Hmm one workaround for now I can think of is to read the files and send them directly to the iframe. Alternatively you could embed them into the HTML. Both approaches are kinda tedious though.

Hello! The creator of Markmind has developed a remarkable product. They have implemented a feature that allows specific PDF annotations to link and jump to an Obsidian markdown page. I believe Markmind excellently complements Obsidian's functionality with PDFs. It would be unfortunate if, due to certain security updates, this product could no longer function fully.

**Comment by lishid** [comment]
I think given most people are using some form of plugins it will offer zero protection for most people. I do have a potential solution, assuming you can load your iframe resources from the same folder as the frame html.

**Comment by MarkMindCkm** [comment]
> I think given most people are using some form of plugins it will offer zero protection for most people. I do have a potential solution, assuming you can load your iframe resources from the same folder as the frame html.

That's great, how can i do it ?

**Comment by lishid** [comment]
Unfortunately I tried a few things and they all turned out to have vulnerabilities or various ways that can be used to bypass. That means you'll need to find a way to embed your javascript and css files into the html directly... Sorry about that.

**Comment by MarkMindCkm** [comment]
> Unfortunately I tried a few things and they all turned out to have vulnerabilities or various ways that can be used to bypass. That means you'll need to find a way to embed your javascript and css files into the html directly... Sorry about that.

ok

**Comment by cirida** [comment]
I have this issue as well! How do you work around this? I have a bunch of plotly graphs I've been displaying this way...

**Comment by MarkMindCkm** [comment]
> I have this issue as well! How do you work around this? I have a bunch of plotly graphs I've been displaying this way...

I have no idea ， If users can choose whether to enable this feature, that would be great

**Comment by vskorjanc** [comment]
> I have this issue as well! How do you work around this? I have a bunch of plotly graphs I've been displaying this way...

Same here ✌

**Comment by calandradas** [comment]
the same issue, have any solution? thanks !

**Comment by lishid** [comment]
For now, the only solution seems to be bundling all of your CSS and JS into the HTML file that you are trying to distribute.

Until we can find a better way in Electron to make sure that pages can't access resources out of their folders I don't think we can safely allow this to happen.

**Comment by idewindy** [comment]
I'm using the singleFile browser plugin to crop it into an html how do I embed it

**Comment by zfyangc** [comment]
> Yeah, we had to block these because they were a security vulnerability. What URL/origin is your iframe using?

ok, can you tell me which obsidian version still supports iframe? I'm looking forward to your reply.

**Comment by vskorjanc** [comment]
> ok, can you tell me which obsidian version still supports iframe? I'm looking forward to your reply.

I believe that the last release that supported iframes was 1.4.16. I downgraded to this version and it works fine for me. You can download old releases [here](https://github.com/obsidianmd/obsidian-releases/releases).

**Comment by justarmadillo** [comment]
> For now, the only solution seems to be bundling all of your CSS and JS into the HTML file that you are trying to distribute.
> 
> Until we can find a better way in Electron to make sure that pages can't access resources out of their folders I don't think we can safely allow this to happen.

MarkMind plugin is a revolutionary plugin for obsidian, you are killing it ...
